src/clarity-angular/select/option.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {AfterViewInit, Component, ElementRef, HostListener, Input, OnDestroy, ViewChild} from "@angular/core";
+import {Subscription} from "rxjs/Subscription";
+
+import {DetailWrapper} from "./../../app/datagrid/expandable-rows/detail-wrapper";
+import {ListsCompactDemo} from "./../../app/lists/lists-compact";
+import {RootSelectService} from "./providers/select.service";
+import {Select} from "./select";
+
+@Component({
+    selector: "clr-option",
+    template: `
+    <li *ngIf="visible" [class.highlight]="highlight" [class.selected]="selected">
+        <div class="wrapper" #contentWrapper>
+            <ng-content></ng-content>
+        </div>
+        <div class="wrapper" #contentWrapper *ngIf="contentWrapper.children.length == 0">
+            {{clrTitle}}
+        </div>  
+    </li>
+    `
+})
+export class Option implements OnDestroy, AfterViewInit {
+    private _value: String;
+    private _title: String;
+    private _visible: boolean = true;
+    private _highlight: boolean = false;
+    private _selected: boolean = false;
+    private _subHighlighted: Subscription;
+    private _subSelected: Subscription;
+    private _subInput: Subscription;  // TODO: implement bold as hint
+    private _index: number;
+    constructor(public selectService: RootSelectService, public el: ElementRef) {}
+
+    ngAfterViewInit() {
+        this._subHighlighted = this.selectService.highlightedChange.subscribe((option: Option) => {
+            if (option && option === this) {
+                this.highlight = true;
+            } else {
+                this.highlight = false;
+            }
+            this.el.nativeElement.scrollTop = 10;
```

Comment:
my mistake, it was a test how to handle the height fix of 10 elements because the problem with limiting results to 10 is now you have a select menu with a fixed height which then with `overflow-y: auto` has a scrollbar which works by mouse quite well but as we support going through result by arrow keys that raises a real problem and with this line i was testing if we could manually scroll every time we come close to an option at the end of the currently viewable options.